cmd/health_checker/main.go

```diff
@@ -0,0 +1,386 @@
+		tlsConfig.RootCAs = rootCAs
+	}
+
+	return tlsConfig, nil
+}
+
+func createHTTPClient(v *viper.Viper, log *zap.Logger) (*http.Client, error) {
+
+	verbose := v.GetBool("verbose")
+
+	clientKeyEncoded := v.GetString("key")
+	clientCertEncoded := v.GetString("cert")
+	skipVerify := v.GetBool("skip-verify")
+
+	if verbose {
+		if skipVerify {
+			log.Info("Skipping client-side certificate validation")
+		}
+	}
+
+	tlsConfig := &tls.Config{}
+
+	if len(clientKeyEncoded) > 0 && len(clientCertEncoded) > 0 {
+
+		clientKey, err := base64.StdEncoding.DecodeString(clientKeyEncoded)
+		if err != nil {
+			return nil, errors.Wrap(err, "error decoding client key")
+		}
+
+		clientCert, err := base64.StdEncoding.DecodeString(clientCertEncoded)
+		if err != nil {
+			return nil, errors.Wrap(err, "error decoding client cert")
+		}
+
+		caBytes := make([]byte, 0)
+		if caEncoded := v.GetString("ca"); len(caEncoded) > 0 {
+			caString, err := base64.StdEncoding.DecodeString(caEncoded)
+			if err != nil {
+				return nil, errors.Wrap(err, "error decoding certificate authority")
+			}
+			caBytes = []byte(caString)
+		}
+
+		tlsConfig, err = createTLSConfig([]byte(clientKey), []byte(clientCert), caBytes, false)
+		if err != nil {
+			return nil, errors.Wrap(err, "error creating TLS config")
+		}
+
+	} else {
+
+		clientKeyFile := v.GetString("key-file")
+		clientCertFile := v.GetString("cert-file")
+
+		if len(clientKeyFile) > 0 && len(clientCertFile) > 0 {
+
+			clientKey, err := ioutil.ReadFile(clientKeyFile) // #nosec b/c we need to read a file from a user-defined path
+			if err != nil {
+				return nil, errors.Wrap(err, "error reading client key file at "+clientKeyFile)
+			}
+
+			clientCert, err := ioutil.ReadFile(clientCertFile) // #nosec b/c we need to read a file from a user-defined path
+			if err != nil {
+				return nil, errors.Wrap(err, "error reading client cert file at "+clientKeyFile)
+			}
+
+			caBytes := make([]byte, 0)
+			if caFile := v.GetString("ca-file"); len(caFile) > 0 {
+				content, err := ioutil.ReadFile(caFile) // #nosec b/c we need to read a file from a user-defined path
+				if err != nil {
+					return nil, errors.Wrap(err, "error reading ca file at "+caFile)
+				}
+				caBytes = content
+			}
+
+			tlsConfig, err = createTLSConfig(clientKey, clientCert, caBytes, false)
+			if err != nil {
+				return nil, errors.Wrap(err, "error creating TLS config")
+			}
+		}
+	}
+
+	httpTransport := &http.Transport{TLSClientConfig: tlsConfig}
+	httpClient := &http.Client{Transport: httpTransport}
+	return httpClient, nil
+
+}
+
+func checkURL(httpClient *http.Client, url string, validStatusCodes intSlice, maxTries int, backoff int, log *zap.Logger) error {
+	for tries := 0; tries < maxTries; tries++ {
+		resp, err := httpClient.Get(url)
+		if err != nil {
+			return errors.Wrap(err, "error calling GET on url "+url)
+		}
+		log.Info(
+			"on try "+strconv.Itoa(tries)+", url "+url+" returned status code "+strconv.Itoa(resp.StatusCode),
```

Comment:
Since you already put the `tries`, `url`, and `resp.StatusCode` in the zap fields, you can remove them from the main log line. 
```suggestion
			"request completed", 
			zap.Int("try", tries),
			zap.String("url", url),
			zap.Int("code", resp.StatusCode))
```